servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/SpliceFlatStreamToPackedSingle.java

```diff
@@ -229,10 +218,10 @@ public void onNext(@Nullable Object obj) {
         }
 
         @Nonnull
-        private Publisher<Payload> newPayloadPublisher() {
-            return new SubscribablePublisher<Payload>() {
+        private Publisher<T> newPayloadPublisher() {
+            return new Publisher<T>() {
```

Comment:
It will be beneficial (helps to avoid allocations when conversion is performed) if we can preserve `implements PublisherSource<T>`. Even if `SubscribablePublisher` is not available in this module, we can duplicate it here.

@bryce-anderson how do you think, from the context propagation point of view, should we use `AbstractSynchronousPublisher` here instead of `SubscribablePublisher`?

Reply:
tbh, the structure there isn't 100% clear to me, but it looks like `AbstractSynchronousPublisher` has the behavior we want even if the name may not make sense in in this case.

Reply:
Added `AbstractSynchronousPublisher`